[Bug] [Seatunnel-dist] Duplicate maven dependency caused mvnd compile failure

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

seatunnel-dist/pom.xml contains a duplicate dependency, thus lead to the build failure. 
```
 The project org.apache.seatunnel:seatunnel-dist:2.3.12-SNAPSHOT (/Users//IdeaProjects/seatunnel-fork/seatunnel-dist/pom.xml) has 1 error
[ERROR]     'profiles.profile[seatunnel].dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.seatunnel:connector-doris:jar -> duplicate declaration of version ${project.version} @ org.apache.seatunnel:seatunnel-dist:${revision}, file:///Users//IdeaProjects/seatunnel-fork/seatunnel-dist/pom.xml, line 493, column 17

````
                <dependency>
                    <groupId>org.apache.seatunnel</groupId>
                    <artifactId>connector-doris</artifactId>
                    <version>${project.version}</version>
                    <scope>provided</scope>
                </dependency>

### SeaTunnel Version

2.3.11

### SeaTunnel Config

```conf
none
```

### Running Command

```shell
none
```

### Error Exception

```log
The project org.apache.seatunnel:seatunnel-dist:2.3.12-SNAPSHOT (/Users//IdeaProjects/seatunnel-fork/seatunnel-dist/pom.xml) has 1 error
[ERROR]     'profiles.profile[seatunnel].dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.seatunnel:connector-doris:jar -> duplicate declaration of version ${project.version} @ org.apache.seatunnel:seatunnel-dist:${revision}, file:///Users//IdeaProjects/seatunnel-fork/seatunnel-dist/pom.xml, line 493, column 17
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-dist/pom.xml
+++ b/seatunnel-dist/pom.xml
@@ -450,12 +450,6 @@
                 </dependency>
                 <dependency>
                     <groupId>org.apache.seatunnel</groupId>
-                    <artifactId>connector-doris</artifactId>
-                    <version>${project.version}</version>
-                    <scope>provided</scope>
-                </dependency>
-                <dependency>
-                    <groupId>org.apache.seatunnel</groupId>
                     <artifactId>connector-slack</artifactId>
                     <version>${project.version}</version>
                     <scope>provided</scope>

